### ISSUE TITLE
MIDI MSC

### ISSUE THREAD

**Issue by offtools** [FrancescoCeruti/linux-show-player#59]
Before posting an issue:
 - avoid opening a new issue if you need a simple or a yes/no answer, use `gitter` instead
 - check if the same issue is already been reported
 - check if the enhancement is already mentioned in the `TODO project`

When writing an issue:
 - use the markdown formatting to make the issue easier to read
 - don't worry about being sarcastic ;-)
 - for bugs/unexpected behavior, use the following "template"
 - remove the template text you don't need, like these first 9 lines

---

Short description of the issue, if the title isn't enough.

**Comment by FrancescoCeruti** [comment]
Yeah, no harm in adding it to the TODO, it seems quite useful and widely adopted.

Maybe a specific cue for them? The input can be added to the `controller` (this should support "global" controls in 0.4.3)?

This make me think, can be useful a plugin that allow to create some kind of template (to match messages) to globally control the cues, easily, without the need to setup every cue? Eventually supporting both receiving and sending messages with various protocols (MIDI/MSC/OSC/...).

**Comment by offtools** [comment]
Hi,

For Outgoing MSC a own cue type make sense, as the argument size is variable in difference to the standard MIDI messages. As the possibities of MSC and the range of the commands is by around a number of over 50 it makes also sense to put that in an own cuetype.

Asuming global controls means things like controls which are not configured inside a Cue, e.g. Go/Stop/Play in the cuelist. For OSC these global (/lisp/go ...) incoming messages are handled by native OSC callbacks, the rest is forwared to the controls via the fallback callback. A unified interface for that would be useful: so that the user could setup: GO is triggered by Osc message  XY and Midi message XZ as it could be done now for the GO keysequence.

**Comment by FrancescoCeruti** [comment]
> Asuming global controls means things like controls which are not configured inside a Cue, e.g. Go/Stop/Play in the cuelist. For OSC these global (/lisp/go ...) incoming messages are handled by native OSC callbacks, the rest is forwared to the controls via the fallback callback. A unified interface for that would be useful: so that the user could setup: GO is triggered by Osc message XY and Midi message XZ as it could be done now for the GO keysequence.

Exactly, the idea is to provide it as part of the `controller` plugin, if it's simple enough, otherwise as a separate plugin.

**Comment by aroomthedoomed** [comment]
hi @offtools, hi @FrancescoCeruti ,

do you have any news about MSC and OSC implantation ?

is there any code that we can already use ?

**Comment by FrancescoCeruti** [comment]
Hi, an OSC plugin has already been merged into the dev branch, but right now the branch might be unusable,
there is an MSC pull-request (#68) with an implementation, but I don't remember if it's complete.
